### ISSUE TITLE
Why is this reported as no-unused-vars?

### ISSUE THREAD

**Issue by daningalla** [eslint/eslint#12357]
```
for(const entry of map){
     code = code.replace(entry.p, entry.r);
}
```
Line 90:  'entry' is defined but never used  no-unused-vars

Edit:
version 1.13.0
what happened: The following warning was issued: 'Line 90:  'entry' is defined but never used  no-unused-vars'
expected: since `entry` is used in the for loop body, I'd expect to not see that warning

**Comment by platinumazure** [comment]
Do you use `code` somewhere else, after the for-of loop?

If not, the lint message is correct- there's not much value in doing all of those replacements and reassigning to a variable you don't later use.

**Comment by daningalla** [comment]
@platinumazure  The message indicates the unused var is `entry`, which you can see is referenced in the loop body. To your question though, `code` is returned from the function it resides in.

**Comment by platinumazure** [comment]
This is why we ask for all the information about your use case... You didn't even include the error message, so I had to guess. :smile:

Please provide all the info requested in [this comment](https://github.com/eslint/eslint/issues/12357#issuecomment-537279532). Thanks!

**Comment by daningalla** [comment]
@platinumazure  - gotcha, sorry. I edited the oc.

**Comment by mysticatea** [comment]
Hi @daningalla. ESLint 1.13.0 doesn't exist. This is the version list: https://www.npmjs.com/package/eslint?activeTab=versions

Please fill our issue template. People have many varied environments, so we need the information that is to reproduce your case in order to investigate it. Otherwise, what we can do is very limited.

Just for information, we cannot reproduce your case on our [online demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50OiBuby11bnVzZWQtdmFyczogZXJyb3IgKi9cbmZvcihjb25zdCBlbnRyeSBvZiBtYXApe1xuICAgICBjb2RlID0gY29kZS5yZXBsYWNlKGVudHJ5LnAsIGVudHJ5LnIpO1xufSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19).

I remember an old issue https://github.com/eslint/eslint/issues/12117 from your code. Would you check if it's your case?

Thank you.

**Comment by daningalla** [comment]
**Environment (from package-lock):**
node = v10.16.3
npm = 6.11.2
babel/core = 7.5.5
babel-eslint: 10.0.2
eslint: 6.4.0

**Configuration**
The only relevant configuration I could find is:
 ```
 "eslintConfig": {
    "extends": "react-app"
  },
```

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Code in question:
```
    for(const entry of map){
        code = code.replace(entry.p, entry.r);
    }
```
Invoked using
```
react-scripts start
```

**What did you expect to happen?**
No warnings from eslint

**What actually happened? Please include the actual, raw output from ESLint.**
```
./src/components/shared/codeBox.jsx
Line 93:  'entry' is defined but never used  no-unused-vars
```

This is precisely what was going on in #12117. Also sorry if I missed other info, as you can easily tell I'm just starting in the JS space...

**Comment by kaicataldo** [comment]
Have you tried the solution in that other issue? Looks like you might need to update babel-eslint.

**Comment by mysticatea** [comment]
Thank you for filling the template.

Okay, you look using problematic `babel-eslint` 10.0.2. Please update it to `10.0.3`.

**Comment by daningalla** [comment]
Thank you - will close.
